app.rb

```diff
@@ -137,11 +126,65 @@ def list_rental
       print 'Person ID: '
       person_id = gets.chomp.to_i
       puts 'Rentals'
-      @rentals.each_with_index do |rental, index|
-        if rental[index].person.id == person_id
-          puts "Date: #{rental[index].date}, Book: #{rental[index].book.title} by #{rental[index].book.author}"
+      @rentals.each do |rental|
+        if rental.person.id == person_id
+          puts "Date: #{rental.date}, Book: #{rental.book.title} by #{rental.book.author}"
         end
       end
     end
   end
+
+  def save_books(filename = 'books.json')
+    File.write(filename, @books.to_json)
+  end
+
+  def save_people(filename = 'people.json')
+    File.write(filename, @people.to_json)
+  end
+
+  def save_rentals(filename = 'rentals.json')
+    File.write(filename, @rentals.to_json)
+  end
+
+  def save_files
+    save_books
+    save_people
+    save_rentals
+  end
+
+  def load_books(filename = 'books.json')
+    JSON.parse(File.read(filename)).each do |book|
+      @books << Book.new(book['title'], book['author'])
+    end
+  end
```

Comment:
![image](https://user-images.githubusercontent.com/65068771/169156121-de6ef04d-531e-4f4b-91fa-eb80c81e8fbd.png)

- [ ] Kindly note that one of the requirements says that On startup of your app load all data from files, and handle the case when one or more of the files are missing (the app should work and show no errors). Kindly fix the error that occurs when one of the `json` files misses. Hint :point_down:

```rb 
  def load_books(filename = 'books.json')
    books = File.exist?(filename) ? JSON.parse(File.read(filename)) : []
    books.each do |book|
      @books << Book.new(book['title'], book['author'])
    end
  end
```